packages/react-sdk/docusaurus/docs/React/02-tutorials/03-livestream.mdx

```diff
@@ -1,6 +1,339 @@
+        if (call) {
+          setActiveCall(call);
+        }
+      });
+  }, [client, callId]);
+
+  return (
+    <StreamVideo client={client}>
+      <StreamMeeting call={activeCall} autoJoin={true}>
+        <WebRTCLivestreamUI />
+      </StreamMeeting>
+    </StreamVideo>
+  )
+}
+
+const WebRTCLivestreamUI = () => {
+  const call = useCall();
+  const isJoined = useCallCallingState() === CallingState.JOINED;
+  return (
+    <>
+      <ViewerHeader />
+      // use any layout from the SDK or build your own
+      <PaginatedGridLayout />
+
+      // show a loading indicator while the call is joining
+      {!isJoined && <LoadingIndicator />}
+      <ViewerControls />
+    </>
+  );
+};
+
+```
+
+### HLS-based live streaming
+
+For the HLS scenario, we are going to use one very popular library called `hls.js`.
+HLS.js is a JavaScript library that implements an HTTP Live Streaming client. It relies on HTML5 video and MediaSource Extensions for playback.
+You can read more about HLS.js [here.](https://github.com/video-dev/hls.js/)
+
+For the HLS-based live streaming, we are going to use a few components from the Stream Video SDK for React but, instead using a pre-built layout,
+we are going to set up the HLS.js library and use it to play the live stream.
+
+Compared to the WebRTC example, we would just replace the "UI" component with the following:
+
+
+```tsx
+import { ... } from '@stream-io/video-react-sdk';
+import HLS from 'hls.js';
+
+export const ViewerApp = () => {
+  // pseudo-code for simplicity, see the previous example for the full code
+  const client = useCreateStreamVideoClient({...});
+  const [activeCall] = client.queryCalls({...});
+  return (
+    <StreamVideo client={client}>
+      <StreamMeeting 
+        call={activeCall} 
+        // we don't want to join this call, as that would switch to WebRTC-based live streaming.
+        // we are going to use HLS.js to play the live stream
+        autoJoin={false}
+      >
+        <HLSLivestreamUI />
+      </StreamMeeting>
+    </StreamVideo>
+  )
+}
+
+export const HLSLivestreamUI = () => {
+  // The HLS playlist URL is in call's metadata
+  // we'll use the `useCallMetadata()` hook to get it
+  const { hls_playlist_url } = useCallMetadata();
+
+  const [isPlaying, setIsPlaying] = useState(false);
+  const videoRef = useRef<HTMLVideoElement>(null);
+  useEffect(() => {
+    if (!call || !hls_playlist_url) return;
+  
+    const hls = new HLS();
+    hls.loadSource(hls_playlist_url);
+    hls.attachMedia(videoRef.current);
+  }, [call]);
+
+  return (
+    <>
+      <ViewerHeader />
+      // this is the video that HLS.js would use to playback the stream.
+      <video autoPlay playsInline ref={videoRef} onPlay={() => setIsPlaying(true)} />
+      
+      {!isPlaying && <LoadingIndicator />}
+      <ViewerControls />
+    </>
+  );
+};
+```
+
+Having this done, we should have a fully functional live streaming experience for the viewers based on HLS.
```

Comment:
Let's be bold here, we SHOULDN'T have it but in fact WE REALLY DO!!!💪🏻

```suggestion
Having done this, we now have a fully functional live-streaming experience for the viewers based on HLS.
```

Reply:
Indeed! Fair point :)